### PR TITLE
docs(guide): environment specific config

### PR DESCRIPTION
**Pull request by eric-horodyski**
# What this PR includes

This pull request contains a guide - _Creating Environment Specific Configurations_ - that walks readers through setting up a QA environment configuration alongside the default environment configuration Capacitor provides out-of-the-box. 

This is achieved by creating an additional iOS scheme in Xcode and setting up Android product flavors in the `build.gradle` file.

The app name and bundle ID differ between the two environment configurations to demonstrate differences between each environment.

### CODE REVIEW

File: pages/docs/v3/guides/environment-specific-configurations.md [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+You will need a Capacitor app with both iOS and Android platforms added. If you have an existing Capacitor app with both platforms added, skip this section.
+
+Depending on your preference, you can either [add Capacitor to an existing web application](/docs/getting-started) or [create a new Capacitor application with the Ionic Framework](/docs/getting-started/with-ionic).
+
+The Capacitor app must use TypeScript for configuration. This guide uses `capacitor.config.ts` to dynamically export different configurations.
+
+Before adding any native platforms to the project, you must build the Capacitor App at least once.
+
+```bash
+npx run build
```
Comment (kensodemann) [inline pages/docs/v3/guides/environment-specific-configurations.md:29]: Shouldn't that be `npm run build`?

Comment (eric-horodyski) [reply pages/docs/v3/guides/environment-specific-configurations.md:29]: Yes, it should...thanks for the find! 😁

File: pages/docs/v3/guides/environment-specific-configurations.md [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+You can find additional information on iOS targets [at this link](https://developer.apple.com/library/archive/documentation/ToolsLanguages/Conceptual/Xcode_Overview/WorkingwithTargets.html).
+
+### Rename the new scheme and Plist file
+
+1. Select **Manage Schemes...** from the Scheme menu.
+2. Find the "App copy" scheme and press the `Enter` key to rename it. Set the name to "App QA" and close the dialog.
+3. Find the "App copy-Info" file in the Project Navigator panel and press the `Enter` key to rename it. Set the file's name to "App QA-Info.plist".
+4. Return to the project's settings. Ensuring that the "App QA" target is selected, open the _Build Settings_ section. Scroll down to Packaging and change the **Info.plist File** entry to "App QA-Info.plist".
+
+The iOS project now has two runnable schemes: "App" and "App QA". Capacitor's configuration file allows you to supply which scheme to build during the `open` and `run` commands.
```
Comment (jcesarmobile) [inline pages/docs/v3/guides/environment-specific-configurations.md:60]: the scheme is not used on open command, open command just opens the workspace

File: pages/docs/v3/guides/environment-specific-configurations.md [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+
+Add the property below to the configuration object in `capacitor.config.ts`:
+
+```typescript
+ios: {
+  scheme: 'App QA',
+}
+```
+
+The `scheme` property tells Capacitor which iOS scheme to use for the `open` and `run` commands. Test this out; run `npx cap run ios` and you'll see that the app name is different.
```
Comment (jcesarmobile) [inline pages/docs/v3/guides/environment-specific-configurations.md:103]: same, not used for open

File: pages/docs/v3/guides/environment-specific-configurations.md [hunk 4]
```diff
@@ -0,0 +1,236 @@
...
+
+Add the property below to the configuration object in `capacitor.config.ts`:
+
+```typescript
+android: {
+   flavor: "qa",
+ },
+```
+
+Test this out; run `npx cap run open` and you'll see that the app name is different.
```
Comment (jcesarmobile) [inline pages/docs/v3/guides/environment-specific-configurations.md:164]: did you mean npx cap run android?
```suggestion
Test this out; run `npx cap run android` and you'll see that the app name is different.
```

**Review by jcesarmobile — CHANGES_REQUESTED** [review]

File: pages/docs/v3/guides/environment-specific-configurations.md [hunk 5]
```diff
@@ -0,0 +1,236 @@
...
+Before adding any native platforms to the project, you must build the Capacitor App at least once.
+
+```bash
+npx run build
+```
+
+Once built, you can add the platforms.
+
+```bash
+npm install @capacitor/{ios,android}
```
Comment (elylucas) [inline pages/docs/v3/guides/environment-specific-configurations.md:35]: might be good to actually fill out the ios/android if people try to copy/past the commands

Comment (elylucas) [reply pages/docs/v3/guides/environment-specific-configurations.md:35]: NM, TIL that you can install multiple packages that way 😂

**Review by elylucas — APPROVED** [review]
Great guide. Two things that came up for me when reading it are how to get different image assets into the apps, and how to deal with runtime environment config like api urls. I think its good to ship now, but maybe some ideas for future.

File: pages/docs/v3/guides/environment-specific-configurations.md (hunk 2 above)
Comment (eric-horodyski) [reply pages/docs/v3/guides/environment-specific-configurations.md]: Removed, thanks for pointing this out!

**Review by jcesarmobile — APPROVED** [review]

**Pull request merged**